### PR TITLE
export: improve formatting and layout (bug 1777559)

### PR DESCRIPTION
**Pull request by zzzeid**
- improve formatting of owners, peers, and paths
- link people to p.m.o search query
- link paths to searchfox search query
- improve lint issues that came up during ./mach lint
- add owner(s) and peer(s) emeritus fields to export
- add support for linking to searchfox based on repo

### CODE REVIEW

File: src/mots/export.py [hunk 1]
```diff
@@ -29,6 +31,48 @@ def escape_for_rst(value: str) -> str:
...
+        path = path.replace("*", "\\*")
+        if searchfox_base_url:
+            path = f"`{path} <{searchfox_base_url}{path}>`__"
+        parsed_paths.append(path)
+    return f"\n{' ' * indent}| " + f"\n{' ' * indent}| ".join(parsed_paths)
+
+
+def format_people_for_rst(value: list, indent: int = DEFAULT_LIST_TABLE_INDENT) -> str:
+    """Escape and format a list of people."""
+    people_base_url = "https://people.mozilla.org/s?query="
```
Comment (cgsheeh) [inline src/mots/export.py:61]: Nit: this could be defined in the module instead of this function.

**Review by cgsheeh — CHANGES_REQUESTED** [review]

File: src/mots/export.py [hunk 2]
```diff
@@ -29,6 +31,48 @@ def escape_for_rst(value: str) -> str:
...
+    else:
+        if searchfox_enabled:
+            searchfox_base_url = (
+                f"https://searchfox.org/{config['repo']}/search?q=&path="
+            )
+
+    parsed_paths = []
+    for path in value:
+        path = path.replace("*", "\\*")
+        if searchfox_base_url:
```
[the remark this answers is not in the document]
Comment (zzzeid) [reply src/mots/export.py:55]: Reworked this.

File: src/mots/export.py (hunk 1 above)
Comment (zzzeid) [reply src/mots/export.py]: Moved to settings module instead, which is probably more apt.

File: src/mots/export.py [hunk 3]
```diff
@@ -32,20 +35,19 @@ def escape_for_rst(value: str) -> str:
...
     """Escape and format a path string (e.g. for includes or excludes)."""
     config = directory.config_handle.config
     try:
         searchfox_enabled = config["export"]["searchfox_enabled"]
     except KeyError:
         searchfox_enabled = False
-        searchfox_base_url = None
+
+    if searchfox_enabled:
+        searchfox_base_url = f"https://searchfox.org/{config['repo']}/search?q=&path="
```
Comment (cgsheeh) [inline src/mots/export.py:48]: Nit: should we have the searchfox URL in `settings` in the same way as the PMO URL?

**Review by cgsheeh — APPROVED** [review]

**Review by grahamhub — APPROVED** [review]

**Review by cgsheeh — APPROVED** [review]

**Pull request merged**